### PR TITLE
WQ: Abstractions with Chunk Size - Map

### PR DESCRIPTION
**Pull request by bwiseman77**
Conflicts:
	work_queue/src/bindings/python3/work_queue.binding.py

### CODE REVIEW

**Comment by bwiseman77** [comment]
Not sure What I should do with other PR, so I will just leave both open for now? git is confusing and I dont wanna mess anything else up

**Comment by bwiseman77** [comment]
todo: 
draw/think about tree reductions
time map
fix map output

**Comment by dthain** [comment]
@bwiseman77 I think a constructive thing to do here is to replace the old `wq.map` with your new map implementation (just call it `wq.map`), and complete and commit that as a single PR and call it done.

Then, you can start thinking about how to improve `wq.treereduce` and then make that a new PR for consideration.  Likewise for `wq.allpairs`.

**Comment by bwiseman77** [comment]
Ok, I just replaced map on this PR, is that ok? Should I close the other PR?

**Comment by dthain** [comment]
Yes, close the other one if you don't need it any more.

File: work_queue/src/bindings/python3/work_queue.binding.py [hunk 1]
```diff
@@ -1745,19 +1746,30 @@ def wait(self, timeout=WORK_QUEUE_WAITFORTASK):
     # @param self       Reference to the current work queue object.
     # @param fn         The function that will be called on each element
     # @param seq        The sequence that will call the function
-    def map(self, fn, seq):
-        results = [None] * len(seq)
+    # @param chunk_size The number of elements to process at once
+
+    def Nmap(self, fn, array, chunk_size=1):
```
Comment (dthain) [inline work_queue/src/bindings/python3/work_queue.binding.py:1751]: Call this one `map` since it's a drop in replacement for the old one.

Comment (bwiseman77) [reply work_queue/src/bindings/python3/work_queue.binding.py:1751]: Op yes sorry I meant to but forgot to delete the N.

**Comment by dthain** [comment]
Looks good!

**Pull request merged**